_posts/2021-09-23-unused-beans.adoc

```diff
@@ -0,0 +1,73 @@
+---
+layout: post
+title: 'Unused Beans and Why We Remove Them'
+date: 2021-09-23
+tags: cdi arc 
+synopsis: Why does Quarkus detect and remove unused beans by default?
+author: mkouba
+---
+
+Quarkus is a build-time oriented stack, i.e. it tries to do as much as possible at build time to improve the startup time and memory usage of your application.
+However, it's not always so easy and straightforward.
+Many existing frameworks and libraries do not take this design choice into account.
+And CDI is one of them.
+
+== The Why
```

Comment:
This kind of title is IMO nice if your other titles are in a similar fashion (e.g. The What, The Why, The How....) which isn't the case here.
You could use something like `Motivation` or `Why Bother`.

Reply:
I like "Why Bother".